### ISSUE TITLE
Key bindinds dialog: Broom symbol saves setting even when pressing "cancel"

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#7959]
JabRef version of today on Windows

- [x] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/main and the problem persists

1. ![grafik](https://user-images.githubusercontent.com/1366654/127908478-c790099b-096b-4dee-8b28-bc8ae8aad4e4.png)
2. ![grafik](https://user-images.githubusercontent.com/1366654/127908501-91fe969b-8f7f-4d26-a286-f8fddb33d787.png)
3. Reopen dialog: Keybinding is cleared: ![grafik](https://user-images.githubusercontent.com/1366654/127908543-740b26ab-e233-41ab-b3a6-e6c833e425ef.png)

**Comment by tobiasdiez** [comment]
Maybe somewhat controversial idea: Make all preference changes persistent immediately without the need to hit the "save" button. The "save your changes" strategy feels a bit old-style. Would also make the code considerably more simple since you could simply bind an option to a javafx binding that sets / gets directly in the preference object.

**Comment by Siedlerchr** [comment]
I hate these automatically saving things, especially if there is no undo or you accidentally clicked somewhere. And I, personally,  always feel like: Are my changes really saved?
Some discussions:
https://ux.stackexchange.com/questions/2754/autosave-ui-pattern 
https://ux.stackexchange.com/a/109174

**Comment by tobiasdiez** [comment]
Yeah, as I said, somewhat controversial topic. However, it seems like most applications go towards this auto-save of preferences. At least the following apps do:
- VS Code
- Firefox
- Modern MS Office apps

So the user expectation might go towards always safe.

**Comment by Santoshkrishna-code** [comment]
Hi, I would like to work on this issue. Could you please assign it to me?
